plugins/modules/idrac_license.py

```diff
@@ -0,0 +1,577 @@
+            else:
+                module.fail_json(msg=FAILURE_MSG.format(operation=share_type, license_id=license_id), changed=False, job_details=job_status)
+        else:
+            if status in [200, 202]:
+                module.exit_json(msg=SUCCESS_EXPORT_MSG, changed=True)
+            else:
+                module.fail_json(msg=FAILURE_MSG.format(operation=share_type, license_id=license_id), changed=False)
+        return export_license_status
+
+
+def get_job_status(self, export_license_status):
+    res_uri = validate_and_get_first_resource_id_uri(
+        self.module, self.idrac, SYSTEMS_URI)
+    job_dict = {}
+    job_tracking_uri = export_license_status.headers.get("Location")
+    job_id = job_tracking_uri.split("/")[-1]
+    job_uri = IDRAC_JOB_URI.format(job_id=job_id, res_uri=res_uri[0])
+    job_resp = self.idrac.invoke_request(job_uri, 'GET')
+    job_dict = job_resp.json_data
+    job_dict = remove_key(job_dict, regex_pattern='(.*?)@odata')
+    return job_dict
+
+
+def export_license_local(self, module):
+    uri = validate_and_get_first_resource_id_uri(
+        self.module, self.idrac, SYSTEMS_URI)
+    export_license_url = f"{uri[0]}/Oem/Dell/DellLicenseManagementService/Actions/DellLicenseManagementService.ExportLicense"
+    payload = {}
+    payload["EntitlementID"] = module.params.get('license_id')
+    path = module.params.get('share_parameters').get('share_name')
+    if not (os.path.exists(path) or os.path.isdir(path)):
+        module.exit_json(msg=f"Provided directory path '{path}' is not valid.", failed=True)
+    if not os.access(path, os.W_OK):
+        module.exit_json(msg=f"Provided directory path '{path}' is not writable. Please check if you "
+                             "have appropriate permissions.", failed=True)
+    license_name = module.params.get('share_parameters').get('file_name')
+    if license_name:
+        license_file_name = f"{license_name}_iDRAC_license.txt"
+    else:
+        license_file_name = f"{module.params['license_id']}_iDRAC_license.txt"
+    license_status = self.idrac.invoke_request(export_license_url, "POST", data=payload)
+    license_data = license_status.json_data
+    license_file = license_data.get("LicenseFile")
+    file_name = os.path.join(path, license_file_name)
+    with open(file_name, "w") as fp:
+        fp.writelines(license_file)
+    return license_status
+
+
+def export_license_http(self, module):
+    uri = validate_and_get_first_resource_id_uri(
+        self.module, self.idrac, SYSTEMS_URI)
+    export_license_url = f"{uri[0]}/{EXPORT_LICENCE_NETWORK_SHARE_URI}"
+    payload = {}
+    payload["EntitlementID"] = module.params.get('license_id')
+    if module.params.get('share_parameters').get('share_type') == "http":
+        payload["ShareType"] = "HTTP"
+    else:
+        payload["ShareType"] = "HTTPS"
+    payload["IPAddress"] = module.params.get('share_parameters').get('ip_address')
```

Comment:
replace line 423 to 426 with function get_share_details, check at other places too, if can reduce number of duplicate line. Check if dictionary and lambda function help in this